.github/workflows/build_and_run_idf_examples.yml

```
@@ -0,0 +1,86 @@
+name: ESP-IDF USB Device examples
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        idf_ver: ["release-v5.1", "release-v5.2", "release-v5.3", "release-v5.4", "release-v5.5", "latest"]
+      fail-fast: false
```

We are using fail fast in esp-usb workflows. Maybe we can do the same here.
```suggestion
      fail-fast: true
```

---

For fast builds that are run on GitHub servers, such as this one, it is better not to fail fast. Because in case of an build error that is eg. only on release/5.4 I can see that the other branches pass. This makes it easier to identify and fix.

Fail-fast is useful for our self-hosted runners, where running all the tests for CI that has already encountered an error could slow down CI jobs of other developers..